Port MASTG-TEST-0080: Testing Enforced Updating (ios) and MASTG-TEST-0036: Testing Enforced Updating (android)

TheDauntless opened the pull request:
Thank you for submitting a Pull Request to the OWASP MASTG. Please make sure that:

- [x] Your contribution is written in the 2nd person (e.g. you)
- [x] Your contribution is written in an active present form for as much as possible.
- [x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [x] Your contribution has proper formatted markdown and/or code
- [x] Any references to website have been formatted as [TEXT](URL “NAME”)
- [x] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR closes #2988 and closes #2994

TheDauntless committed:
First try

TheDauntless commented:
The Evaluation doesn't give 100% certainty, so I don't know how to write this down. Do we add a disclaimer?

sydseter commented on tests-beta/android/MASVS-CODE/MASTG-TEST-0x80.md:
```diff
@@ -0,0 +1,26 @@
+## Steps
+
+1. Examine the startup flow of the application. Identify if the application calls out to a backend with the application's version information included.
+2. Examine if the application can handle a specific response from the backend indicating that the application must be updated. For example, the application might evaluate the response from the backend and show a specific error message. Note that the error message can also come from the backend, so the lack of a custom error message in the application does not mean that the feature isn't implement ...
```
Perhaps this part should be clarified? Perhaps it could be written like this:

2. Examine if the application can handle a specific response from the backend indicating that the application must be updated. For example by showing or evaluating an error message retrieved from an API or the mobile application.

cpholguera replied:
I've updated everything in case you'd like to take a look.

sydseter commented on tests-beta/ios/MASVS-CODE/MASVS-TEST-0x80-3.md:
```diff
@@ -0,0 +1,26 @@
+## Steps
+
+1. Examine the startup flow of the application. Identify if the application calls out to a backend with the application's version information included.
+2. Examine if the application can handle a specific response from the backend indicating that the application must be updated. For example, the application might evaluate the response from the backend and show a specific error message. Note that the error message can also come from the backend, so the lack of a custom error message in the application does not mean that the feature isn't implement ...
```
Same as above. Regardless of my comment, the note you are using is a bit confusing to me. Otherwise, I like very much the rest.

ScreaMy7 commented:
@cpholguera i can review the pull request, pls assign.

TheDauntless committed:
Process some comments

cpholguera commented on tests-beta/ios/MASVS-CODE/MASVS-TEST-0x80-2.md:
```diff
@@ -0,0 +1,28 @@
+
+## Observation
+
+The server responds differently to older versions.
```
That's not correct, see above.

cpholguera commented on tests-beta/ios/MASVS-CODE/MASVS-TEST-0x80-3.md:
```diff
@@ -0,0 +1,26 @@
+
+## Evaluation
+
+The test case fails if the application does not contain any logic to handle a forced application update. Additionally, the test case fails if the application informs the user that they must update, but the user can ignore the prompt and still use the application.
```
Not correct, see other related comments.

cpholguera committed:
add deprecation notes

cpholguera committed:
fix new tests metadata

cpholguera committed:
Merge branch 'master' of https://github.com/OWASP/owasp-mastg into MASTG-TEST-0080

cpholguera committed:
add L2 profile metadata

cpholguera committed:
[COPILOT] refine enforced updating tests for Android and iOS

- Clarify Overview sections: include Play In-App Updates details for Android and CFBundle/version retrieval for iOS.
- Expand and standardize Steps: add specific static and dynamic analysis guidance (MITM, dynamic instrumentation, tech refs).
- Improve Observation: list expected artifacts (network trace, method trace),
- Improve Evaluation: spell out failure criteria, and add "Additional Verification" steps for manual checks.
- Normalize wording, fix typos, and align examples across Android and iOS test files.

cpholguera committed:
[COPILOT] Enhance iOS enforced-updating tests to cover iTunes lookup and StoreKit

- Add iTunes Search API lookup checks and parsing (resultCount, results[0].version, results[0].trackViewUrl) to dynamic and static tests
- Expand instrumentation guidance to include parsing logic, UIApplication.open, and SKStoreProductViewController
- Improve observations and verification steps (include lookup stubbing/replay and iTunes lookup-related evaluation)

cpholguera committed:
[COPILOT] Clarify iOS enforced updating guidance and add App Store Lookup / StoreKit details

- Rewrote guidance to explain iOS has no built-in enforced-update API and outline developer responsibilities
- Added App Store Lookup API example endpoint and description of how to compare versions
- Documented SKStoreProductViewController and App Store URL alternatives for update UX
- Mentioned common open-source libraries and linked to App Store Connect / Review Guidelines
- Emphasized backend/API lifecycle considerations and testing of older app versions

cpholguera committed:
[COPILOT] Expand Enforced Updating guidance — prefer Play In‑App Updates API; add AppUpdateManager, update modes, usage, and backend-gated flow recommendations

cpholguera commented on knowledge/android/MASVS-PLATFORM/MASTG-KNOW-0023.md:
```diff
@@ -4,8 +4,29 @@ platform: android
+
+Keep in mind that updating the app does not resolve vulnerabilities residing on backend systems. A secure update mechanism should complement proper API and service lifecycle management. Similarly, if users are not forced to update, test older app versions against your backend and apply API versioning and deprecation policies to maintain security and stability across all supported releases.
+
+## Google Play In‑App Updates API
```
Also add https://support.google.com/googleplay/android-developer/answer/13812041

cpholguera committed:
Merge branch 'master' into MASTG-TEST-0080

cpholguera committed:
Merge branch 'master' into MASTG-TEST-0080

cpholguera committed:
content udpates

cpholguera committed:
update fake IDs

cpholguera committed:
add Play Console recovery tools

cpholguera committed:
Add fixes for DEVELOPER_TRIGGERED_UPDATE_IN_PROGRESS

cpholguera commented on tests/android/MASVS-CODE/MASTG-TEST-0036.md:
```diff
@@ -8,6 +8,9 @@ title: Testing Enforced Updating
 - L2
 profiles: [L2]
+status: deprecated
+covered_by: ['MASTG-TEST-0x80', 'MASTG-TEST-0x80-1']
```
```suggestion
covered_by: ['MASTG-TEST-0x01', 'MASTG-TEST-0x02']
```

cpholguera commented on tests/ios/MASVS-CODE/MASTG-TEST-0080.md:
```diff
@@ -8,6 +8,9 @@ title: Testing Enforced Updating
 - L2
 profiles: [L2]
+status: deprecated
+covered_by: ['MASTG-TEST-0x80-2', 'MASTG-TEST-0x80-3']
```
```suggestion
covered_by: ['MASTG-TEST-0x03', 'MASTG-TEST-0x04']
```

cpholguera committed:
Apply suggestions from code review

Co-authored-by: Carlos Holguera <perezholguera@gmail.com>

cpholguera committed:
Merge branch 'master' into MASTG-TEST-0080

TheDauntless committed:
Small fixes

TheDauntless committed:
fix linting

cpholguera committed:
Merge branch 'master' into MASTG-TEST-0080

cpholguera committed:
fix IDs

cpholguera committed:
fix decommissioned typo
